### ISSUE TITLE
Can't get didFailToRegisterForRemote to be called

### ISSUE THREAD

**Issue by mmdumi** [urbanairship/ios-library#114]
Hi guys,
I don't really know what callback should be sent when "Don't allow" is selected on allow notification popup.
I've tried setting automaticSetupEnabled = false and override didFailToRegisterForRemoteNotificationsWithError in application delegate but it's still not being called since it's swizzled by urban airship implementation.
I'm running version 8.0.0 of the sdk.

**Comment by rlepinski** [comment]
`didFailToRegisterForRemoteNotificationsWithError:` is only called if apns registration fails. Starting with iOS 8 Apple broke out user notifications from apns registration to allow sending data messages to the application even if the user opts out of push. I would expect that method to fail only if your provisioning profile is messed up. What info are you looking for?

**Comment by mmdumi** [comment]
I would need to have a callback informing me that user selected "Don't Allow" on the Apple's push notifications alert.
Right now I can only know that user selected "Allow" since I'm receiving the push token.

**Comment by rlepinski** [comment]
For iOS 8 & iOS 9 you can check the user notification settings on the application and for iOS 10 you have to check the `UNUserNotificationCenter`. Alternatively you can take a look at `[UAirship push].authorizedNotificationOptions` which will be 0 if the user opted out of push. We will look into exposing a delegate method that notifies the app on notification opt-in in the future.

**Comment by mmdumi** [comment]
That would be great. Thanks.
For this project I'm working on I really need to know how the user responded to the allow push popup because the next flows are different.

Currently as a workaround I've seen that applicationDidBecomeActive is called after user makes his choice and only then I can read the state how you suggested. I'm wondering why there isn't a better way to do this.

**Comment by rlepinski** [comment]
Released 8.1.0 with new two new methods in the UARegistrationDelegate:
```
/**
 * Called when APNS registration completes.
 *
 * @param options UANotificationOptions that were most recently registered.
 * @param categories NSSet of the categories that were most recently registered.
 */
- (void)notificationRegistrationFinishedWithOptions:(UANotificationOptions)options categories:(NSSet *)categories;

/**
 * Called when APNS authentication changes with the new authorized options.
 *
 * @param options UANotificationOptions that were most recently registered.
 */
- (void)notificationAuthorizedOptionsDidChange:(UANotificationOptions)options;
```

The one you will care about is `notificationAuthorizedOptionsDidChange:` which will be called anytime we detect a change in what notification options are enabled by the user.
